common/djangoapps/student/models.py

```diff
@@ -1569,7 +1569,11 @@ def unenroll_by_email(cls, email, course_id):
         """
         try:
             user = User.objects.get(email=email)
-            return cls.unenroll(user, course_id)
+            record = cls.objects.get(user=user, course_id=course_id)
+            if record.refundable():
```

Comment:
If the user's enrollment is `refundable` why are we skipping the refund on the update enrollment?